### PR TITLE
Add a newer Diet Go Go set (with alt graphics)

### PR DESCRIPTION
**Pull request by clawgrip**
New working systems
-------------------
Diet Go Go (Europe v1.1 1992.09.26 v4) [Tailsnic Retroworks]

### CODE REVIEW

**Comment by mamehaze** [comment]
all the European sets will attempt to display the alt graphics on the title screen if you flip the same switch

the problem is they're not in the gfx ROMs for the other Euro sets.

I suspect this is actually a case of the other sets being partial dumps, and it was assumed they used the same video ROMs as the Jp and US versions when in reality they should have these updated ROMs.

**Comment by clawgrip** [comment]
We found the new gfx ROMs on two "Euro V1.1 (v3)" PCBs, but we also have other Euro V1.1 v2/v1 PCBs with MAY-01 and MAY-02 gfx, and factory soldered.
I'd say we should merge V3 and V4 as the same set with the new gfx ROMs, but leaving the others as they are.

**Comment by mamehaze** [comment]
where is 'v3' and 'v4' even coming from? are those official version numbers?

**Comment by clawgrip** [comment]
I'd say it's "set 1", "set 2", etc.

**Comment by mamehaze** [comment]
right, if those aren't official version numbers from the board I'd just merge the first 2 sets then, as only that graphic ROM differs.

it's odd that all the Euro sets upload the data needed for this alt title screen, read the dipswitches, and attempt to draw the sprites (which just happen to be blank tiles in the older GFX ROM revision) if they weren't made with it in mind, but if the older PCBs have been seen without the updated graphics, then yeah, leave those with the old one.

all the Euro sets should have the dipswitch at least though, as the effect is known, it changes the title screen, to either the new graphics, or blank tiles where the new graphics would be depending on which GFX ROM is installed.

File: src/mame/dataeast/dietgo.cpp [hunk 1]
```diff
@@ -322,8 +331,31 @@ void dietgo_state::dietgo(machine_config &config)
...
+	ROM_LOAD( "may-03.11l", 0x00000, 0x80000, CRC(b6e42bae) SHA1(c282cdf7db30fb63340cc609bf00f5ab63a75583) )
+
+	ROM_REGION( 0x0600, "plds", 0 )
+	ROM_LOAD( "pal16l8b_vd-00.6h",  0x0000, 0x0104, NO_DUMP ) // PAL is read protected
+	ROM_LOAD( "pal16l8b_vd-01.7h",  0x0200, 0x0104, NO_DUMP ) // PAL is read protected
+	ROM_LOAD( "pal16r6a_vd-02.11h", 0x0400, 0x0104, NO_DUMP ) // PAL is read protected
+ROM_END
 
-ROM_START( dietgo ) // same version 1.1 and same date as dietgoe but newer version in rom labels
+ROM_START( dietgoe ) // same version 1.1 and same date as dietgoe but newer version in ROM labels
```
Comment (cuavas) [inline src/mame/dataeast/dietgo.cpp:358]: This comment doesn’t make sense any more – “same version 1.1 and same date as dietgoe” when it *is* `dietgoe`.

Please check the rest of the file for things that refer to the different systems by short name when you change short names.

**Comment by happppp** [comment]
Yeah, it looks like the western style title screen gfx works on all European versions. Does it make sense to (ab)use a BIOS setting like so?: https://github.com/mamedev/mame/commit/83f14ed4288d4968c125c460e1764ea2a5f1bfb7

**Comment by mamehaze** [comment]
> Yeah, it looks like the western style title screen gfx works on all European versions. Does it make sense to (ab)use a BIOS setting like so?: [83f14ed](https://github.com/mamedev/mame/commit/83f14ed4288d4968c125c460e1764ea2a5f1bfb7)

I'm not really keen on the BIOS abuse personally

**Comment by happppp** [comment]
Sure, I reverted it.
I'm 50/50 on it for this driver. But here's a more extreme example, where not having a bios setting would result in a quadruple amount of drivers (72 instead of 18): https://github.com/mamedev/mame/blob/master/src/mame/fidelity/elite.cpp

**Comment by clawgrip** [comment]
Is still needed to remove "dietgoe" and assume that the later European ROMs should always use the new sprite ROMs?

**Comment by happppp** [comment]
If jy_00-3+jy_01-3 with may-01+may-02 does not exist

**Comment by clawgrip** [comment]
> If jy_00-3+jy_01-3 with may-01+may-02 does not exist

Cannot tell for sure, I'll leave it as is then.

Thanks!!!!

**Pull request merged**